cmd/tfplugindocs/testdata/scripts/schema-json/migrate/time_provider_success.txtar

```diff
@@ -0,0 +1,909 @@
+}
+-- exp-examples/resources/static/example_2.tf --
+resource "time_static" "ami_update" {
+  triggers = {
+    # Save the time each switch of an AMI id
+    ami_id = data.aws_ami.example.id
+  }
+}
+
+resource "aws_instance" "server" {
+  # Read the AMI id "through" the time_static resource to ensure that
+  # both will change together.
+  ami = time_static.ami_update.triggers.ami_id
+
+  tags = {
+    AmiUpdateTime = time_static.ami_update.rfc3339
+  }
+
+  # ... (other aws_instance arguments) ...
+}
+-- exp-examples/resources/static/import_1.sh --
+$ terraform import time_static.example 2020-02-12T06:36:13Z
+-- exp-templates/index.md.tmpl --
+---
+layout: "time"
+page_title: "Provider: Time"
+description: |-
+  The time provider is used to interact with time-based resources.
+---
+# Time Provider
+
+The time provider is used to interact with time-based resources. The provider itself has no configuration options.
+
+Use the navigation to the left to read about the available resources.
+
+## Resource "Triggers"
+
+Certain time resources, only perform actions during specific lifecycle actions:
+
+- `time_offset`: Saves base timestamp into Terraform state only when created.
+- `time_sleep`: Sleeps when created and/or destroyed.
+- `time_static`: Saves base timestamp into Terraform state only when created.
+
+These resources provide an optional map argument called `triggers` that can be populated with arbitrary key/value pairs. When the keys or values of this argument are updated, Terraform will re-perform the desired action, such as updating the base timestamp or sleeping again.
+
+For example:
+
+{{tffile "$WORK/examples/example_1.tf"}}
+
+`triggers` are *not* treated as sensitive attributes; a value used for `triggers` will be displayed in Terraform UI output as plaintext.
+
+To force a these actions to reoccur without updating `triggers`, the [`terraform taint` command](https://www.terraform.io/docs/commands/taint.html) can be used to produce the action on the next run.
+-- exp-templates/resources/offset.md.tmpl --
+---
+layout: "time"
+page_title: "Time: time_offset"
+description: |-
+  Manages a offset time resource.
+---
+# Resource: time_offset
+
+Manages an offset time resource, which keeps an UTC timestamp stored in the Terraform state that is offset from a locally sourced base timestamp. This prevents perpetual differences caused by using the [`timestamp()` function](https://www.terraform.io/docs/configuration/functions/timestamp.html).
+
+-> Further manipulation of incoming or outgoing values can be accomplished with the [`formatdate()` function](https://www.terraform.io/docs/configuration/functions/formatdate.html) and the [`timeadd()` function](https://www.terraform.io/docs/configuration/functions/timeadd.html).
+
+## Example Usage
+
+### Basic Usage
+
+{{tffile "$WORK/examples/resources/offset/example_1.tf"}}
+
+### Triggers Usage
+
+{{tffile "$WORK/examples/resources/offset/example_2.tf"}}
+
+## Argument Reference
```

Comment:
To further help provider developers with their documentation generation journey, I wonder if we should drop some additional comments into these custom templates, e.g. mentioning `{{ SchemaMarkdown }}` to replace the manual schema documentation if descriptions are added in the provider code and (separately?) mentioning that it may be possible to remove the entire custom template once all resource/schema descriptions are in place.

What do you think?